rust/arrow/src/json/reader.rs

```diff
@@ -157,9 +156,26 @@ fn generate_schema(spec: HashMap<String, HashSet<DataType>>) -> Result<Arc<Schem
 /// `max_read_records` controlling the maximum number of records to read.
 ///
 /// If `max_read_records` is not set, the whole file is read to infer its field types.
-fn infer_json_schema(file: File, max_read_records: Option<usize>) -> Result<Arc<Schema>> {
+pub fn infer_json_schema_from_seekable<R: Read + Seek>(
```

Comment:
May you please add a note on why a user would want/need to use this vs the non-seekable one

Reply:
Updated with additional examples and other tests at the end. could you please check